docs/share_usage.md

```diff
@@ -0,0 +1,611 @@
+The **StorageClass** object defines the storage provisioner (in our 
+case the **HPE 3PAR Volume Plug-in for Docker**) and parameters to be 
+used when requesting persistent storage within a Kubernetes/Openshift 
+environment. The **StorageClass** acts like a "storage profile" and 
+gives the storage admin control over the types and characteristics of 
+the volumes that can be provisioned within the Kubernetes/OpenShift 
+environment. For example, the storage admin can create multiple 
+**StorageClass** profiles that have size restrictions, if they are 
+enabled with ACLs, if a CPG other than the configured one needs to be used etc.
+
+### StorageClass Example<a name="sc"></a>
+
+The following creates a **StorageClass "sc1"** that provisions a 
+default file sharewith the help of HPE 3PAR Docker Volume Plugin.
+
+**Note:** In order to use file share feature, it is mandatory to specify
+'filePersona' option with empty string as value.
+
+```yaml
+$ sudo kubectl create -f - << EOF
+---
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: sc1
+provisioner: hpe.com/hpe
+parameters:
+  filePersona: ""
+EOF
+```
+
+#### Supported StorageClass parameters<a name="sc_parameters"></a>
+
+| StorageClass Options | Type    | Parameters                                 | Example                          |
+|----------------------|---------|--------------------------------------------|----------------------------------|
+| size                 | integer | Size of share in GiB                       | size: "10"                       |
+| cpg                  | String  | Name of the CPG                                   | cpg: SomeCpg             |
+| fpg                  | String  | Existing FPG name including legacy FPG     | fpg: SomeFpg           |
+| fsMode               | String | Unix style permissions or ACL string                                | fsMode: "A:fd:rwa,A:g:rwaxdnNcCoy,A:fdS:DtnNcy"              |
+| fsOwner              | String | User Id and Group Id that should own the mounted directory      | fsOwner: "1000:1000"         
+
+### Persistent Volume Claim Example<a name="pvc"></a>
+
+Now let’s create a claim **PersistentVolumeClaim** (**PVC**). Here we specify the **PVC** name **pvc1** and reference the **StorageClass "sc1"** that we created in the previous step.
+
+```yaml
+$ sudo kubectl create -f - << EOF
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: pvc1
+spec:
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 20Gi
+  storageClassName: sc1
+EOF
+```
+
+At this point, after creating the **SC** and **PVC** definitions, the volume hasn’t been created yet. The actual volume gets created on-the-fly during the pod deployment and volume mount phase.
+
+### Pod Example<a name="pod"></a>
+
+So, let’s create a **pod "pod1"** using the **nginx** container along with some persistent storage:
+
+```yaml
+$ sudo kubectl create -f - << EOF
+---
```

Comment:
Actually the correct POD definition after some analysis is this

```yaml
kind: Pod
apiVersion: v1
metadata:
  name: podfiletestw4-uid-gid-nosecurity
spec:
  containers:
  - name: nginx
    securityContext:
      privileged: true
      capabilities:
        add: ["SYS_ADMIN"]
      allowPrivilegeEscalation: true
    image: nginx
    volumeMounts:
    - name: export
      mountPath: /export
  restartPolicy: Always
  volumes:
  - name: export
    persistentVolumeClaim:
      claimName: sc-file-pvc-uid-gid
```

Reply:
need to merge POD def in the file

Reply:
Okay

Reply:
PR updated accordingly. @wdurairaj - please check. I've updated security-context both in static as well dynamic provisioning sections.